File: hw2/hw2.py
import matplotlib.pyplot as plt
import numpy as np

def get_stepsize(interval, steps):
	return (interval[1] - interval[0])/steps

def forward_euler(funcp, steps, init, interval):
	stepsize = get_stepsize(interval, steps)
	step_set = [init[0] + i*stepsize for i in range(steps)]
	output_set = [init[1]]
	for i, t in enumerate(step_set):
		output_set.append(output_set[i] + stepsize * funcp(t, output_set[i]))
	print(len(output_set[:-1]))
	return step_set, output_set[0:-1]

def run_cases(steps):
	xs, ys = forward_euler(lambda t,y: y, steps, (1,2), (1,4))
	myplot = plt.plot(xs, ys, 'o')#, xs, lambda t: 2*(np.e**(t-1)))
	myplot.show()
	#ode_solver(lambda t,y: -2*t*y, steps, (1,2), (1,4))
	#TODO figure out how to work with the second derivative...

run_cases(1000)

File: hw2/hw2.jl
using Gadfly
using Debug

function stepsize(interval::Tuple{Real, Real}, steps::Int)
	return (interval[2] - interval[1])//steps
end

#Forward Euler
@debug function forwardEuler(fp, steps, init::Tuple{Real, Real}, interval::Tuple{Real, Real})
	stepSet = [init[1] + i*stepsize(interval, steps) for i in 0:steps-1]
	outputSet = Array(Real, size(stepSet))
	outputSet[1] = init[2]
	println(typeof(outputSet))
	for i in 0:size(stepSet)
		outputSet[i+1] = outputSet[i] + stepsize(interval, steps) * fp(t, outputSet[i])
	end
	return stepSet, outputSet
end
#Newton's Method
function newtonsMethod(f)
	for cnt in 1:100
		return 1
	end
end


#Trapezoidal
function trapezoidal(f, steps::Int, init::Tuple{Real, Real}, interval::Tuple{Real, Real})
	vals = [init]
	stepsize = (interval[2] - interval[1])//steps
end

#Midpoint

#RK(2)

#RK(4)

#Plot Solutions

a(t::Real)::Real = 2 * exp(t - 1)
b(t::Real)::Real = 2 * exp(1 - (t ^ 2))
c(t::Real)::Real = sin(t)

#function runCases()
fp1(t, y) = y
	xs, ys = forwardEuler(fp1,	1000,	(1, 2),	(1,4))
	fEuler = plot(x=xs, y=ys)
	draw("fEuler.png", 6inch, 6inch, fEuler)
#end

#runCases()
